controllers/volume/v1alpha1/volume_controller.go

```diff
@@ -624,18 +624,68 @@ func (r *Reconciler) createCNSAttachment(
 	}
 
 	if err := r.Create(ctx, attachment); err != nil {
-		// An IsAlreadyExists error is not really expected because the only way we'll do a Create()
-		// is if the attachment wasn't returned in the List() done at the start of the reconcile.
-		// The Client cache might have been stale, so ignore the error, but like the comment in the
-		// caller, we might want to perform an Update() in this case too.
-		if !apiErrors.IsAlreadyExists(err) {
-			return errors.Wrap(err, "Error creating CnsNodeVmAttachment")
+		if apiErrors.IsAlreadyExists(err) {
+			return r.createCNSAttachmentButAlreadyExists(ctx, attachmentName)
 		}
+		return errors.Wrap(err, "Cannot create CnsNodeVmAttachment")
 	}
 
 	return nil
 }
 
+// createCNSAttachmentButAlreadyExists tries to handle various conditions when a CnsNodeVmAttachment
+// unexpected already exists. Usually the existing attachment is for a prior VC VM has been deleted
+// from underneath us, and the replacement will have a different BiosUUID. The CnsNodeVmAttachment
+// workflow was from CNS and has been very cumbersome to us to use in practice, with this controller
+// being more complicated than it really should be for us to try to work around all the edge cases.
+// Note that CNS doesn't reevaluate an attachment at all once Status.Attached=true. We could Update
+// it instead - clearing Attached atomically since it doesn't have a Status subresource - but it
+// should not be treating the Status as the SoT.
```

Comment:
Can we open a bug/ticket to get this fixed from the CNS team?  Status should not be treated as SoT and should be recalculated during each reconcile.